tests/unit/test_estimator.py

```diff
@@ -191,6 +191,16 @@ def sagemaker_session():
     return sms
 
 
+@pytest.fixture()
```

Comment:
is this per test or per module? (aka are there potential race conditions with the modification of this dictionary in the tests?)

Reply:
The pytest.fixture by default is run once per test function: https://docs.pytest.org/en/stable/fixture.html#scope-sharing-a-fixture-instance-across-tests-in-a-class-module-or-session

Each test is modifying the `RETURNED_JOB_DESCRIPTION.copy()` so should not cause race condition.